common/netlink.cpp

```diff
@@ -0,0 +1,468 @@
+#include "netlink.h"
+
+#include <linux/if_arp.h> // ARPHRD_ defines
+
+#include <unistd.h> // getpid(), getpagesize()
+#include <string.h> // memcpy, memset etc
+
+Netlink::Netlink()
+{
+    m_sequence_number = 1;
+    m_ip = "10.0.3.100";
+    m_gateway = "10.0.3.1";
+    m_broadcast = "10.0.3.255";
+
+    if (!setup_netlink()) {
+        fprintf(stderr, "Failed to setup netlink\n");
+    }
+
+    if (!get_dump()) {
+        fprintf(stderr, "Failed to initialize cache\n");
+    }
+}
+
+Netlink::~Netlink()
+{
+}
+
+bool Netlink::setup_netlink()
+{
+    if ((m_fd = socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE)) == -1) {
```

Comment:
What happens if we run setup_netlink multiple times by accident? Do we need some kind of state?